### PR TITLE
Fix schema validation for CREATE TABLE statements

### LINKED ISSUE #2073: init.sql queries incompatible with schema
https://app.circleci.com/pipelines/github/mozilla/bigquery-etl/9477/workflows/7fc062c9-318a-4349-817e-2c93643d1395/jobs/58484

- `sql/moz-fx-data-shared-prod/telemetry_derived/experiment_search_aggregates_v1/init.sql`
- `sql/moz-fx-data-shared-prod/search_derived/desktop_search_aggregates_by_userstate_v1/init.sql`
- `sql/moz-fx-data-shared-prod/telemetry_derived/experiments_daily_active_clients_v1/init.sql`
- `sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v1/init.sql`

### PR DESCRIPTION
**Pull request by relud**
fixes #2073

### CODE REVIEW

File: bigquery_etl/cli/dryrun.py [hunk 1]
```diff
@@ -94,10 +94,7 @@ def sql_file_valid(sqlfile):
         """Dry run the SQL file."""
         result = DryRun(sqlfile, use_cloud_function=use_cloud_function, client=client)
         if validate_schemas:
-            valid = result.validate_schema()
-            if not valid:
-                click.echo(f"{sqlfile:59} ERROR schema invalid")
-            return valid
+            return result.validate_schema()
```
Comment (relud) [inline bigquery_etl/cli/dryrun.py:97]: `result.validate_schema()` already prints an error message

File: bigquery_etl/dryrun.py [hunk 2]
```diff
@@ -476,11 +476,15 @@ def get_error(self):
...
-            print(f"\t...Ignoring dryrun results for {self.sqlfile}")
+            print(f"\t...Ignoring schema validation for {self.sqlfile}")
             return True
 
         query_file_path = Path(self.sqlfile)
         query_schema = Schema.from_json(self.get_schema())
+        if self.errors():
+            # ignore file when there are errors that self.get_schema() did not raise
+            click.echo(f"\t...Ignoring schema validation for {self.sqlfile}")
+            return True
```
Comment (relud) [inline bigquery_etl/dryrun.py:487]: this is what happens for `CREATE VIEW` and `CREATE TABLE` statements

File: bigquery_etl/schema/__init__.py [hunk 3]
```diff
@@ -127,6 +127,13 @@ def compatible(self, other: "Schema") -> bool:
 
         return True
 
+    @staticmethod
+    def _node_with_mode(node):
+        """Add default value for mode to node."""
+        if "mode" in node:
+            return node
+        return {"mode": "NULLABLE", **node}
```
Comment (relud) [inline bigquery_etl/schema/__init__.py:135]: mode=NULLABLE is missing in the schema returned from dry running a `CREATE TABLE` statement

File: sql/moz-fx-data-shared-prod/telemetry_derived/experiment_search_aggregates_v1/init.sql [hunk 4]
```diff
@@ -1,13 +1,12 @@
 CREATE TABLE IF NOT EXISTS
   `moz-fx-data-shared-prod.telemetry_derived.experiment_search_aggregates_v1`(
-    dataset_id STRING,
```
Comment (relud) [inline sql/moz-fx-data-shared-prod/telemetry_derived/experiment_search_aggregates_v1/init.sql:3]: this query had an out of date schema

**Review by scholtzan — APPROVED** [review]

**Pull request merged**